festim/exports/exports.py

```diff
@@ -8,12 +9,33 @@ class Exports(list):
     """
 
     def __init__(self, *args):
-        super().__init__(*args)
+        # checks that input is list
+        try:
+            super().__init__(*args)
+        except:
+            raise TypeError("festim.Exports must be a list")
+
+        # checks that list elements are festim.Export
+        if len(self) != 0:
+            if not all(
+                isinstance(t, festim.Export) or isinstance(t, festim.DerivedQuantities)
+                for t in self
+            ):
+                raise TypeError("festim.Exports must be a list of festim.Export")
+
         self.t = None
         self.V_DG1 = None
         self.final_time = None
         self.nb_iterations = 0
 
+    @property
+    def exports(self):
+        warnings.warn(
+            "The exports attribute will be deprecated in a future release, please use festim.Exports[:] instead",
+            DeprecationWarning,
+        )
+        return self
+
```

Comment:
I would have a setter/getter for `exports` (same for traps, derived_quantities, etc)

Reply:
As I've checked locally, `my_exports.exports.append()` works fine. Maybe, it's not needed?

Reply:
Ok for append, I suppose that's because we are not setting the attribute but modifying it.

How about:

```python
my_exports.exports = []
```

Running the following throws an error:

```python
class NewList(list):
    @property
    def exports(self):
        return self


my_new_list = NewList([1, 2, 3, 1.2])

my_new_list.exports = [2]
```

Reply:
Right, that won't work